File: Problems/ReverseInteger.php
class Solution {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $data=array();
        if($x<0){
                $flag=1;
                $x=-$x;
            }
        
        for($i=0;$x!=0;$i++){
            $data[$i]=$x%10;
            $x=($x-$data[$i])/10;
        }
        
        for($j=0;$j<$i;$j++)
        {
            $result+=$data[$j]*pow(10,$i-$j-1);
        }
        if($flag==1){
            $result=-$result;
        }
        if($result>(pow(2,31)-1)||$result<-pow(2,31)){
            return 0;
        }
        else
            return $result;  
    }
}

File: InterviewQuestions/reverseInteger.py
class Solution:
    def reverse(self, x: int) -> int:
        x = str(x)
        if x[0] == '-':
            reverseNumber = int('-' + x[-1:0:-1])
            if reverseNumber >= -2147483648 and reverseNumber <= 2147483647:
                return reverseNumber
            else:
                return 0
        else:
            reverseNumber = int(x[::-1])
            if reverseNumber >= -2147483648 and reverseNumber <= 2147483647:
                return reverseNumber
            else:
                return 0
